foundation-ui/auth/AuthService.js

```diff
@@ -0,0 +1,105 @@
+ * import {AuthService} from 'foundation-ui/mount';
+ * AuthService.register('myID', function (value) {
+ *   return value.replace('my', 'myChanged');
+ * });
+ *
+ * AuthPoint would register itself with the AuthService to be able to get
+ * updates on content
+ * import {AuthService} from 'foundation-ui/mount';
+ * function updateContent() {
+ *   let newContent = AuthService.getContent(
+ *     'myID',
+ *     React.Children.toArray(children),
+ *     props
+ *   );
+ *
+ *   // Do something with newContent
+ * }
+ * AuthService.addListener('myID', updateContent);
+ */
+const AuthService = {
+  // Package API
+  /**
+   * Registers package for updates on this AuthPoint permission
+   * @param  {String} permission of MointPoint to register package for
+   * @param  {Function} callback to call when content is being fetched
+   * @param  {Number} priority to register package callback with
+   * [-Infinity; Infinity]
+   */
+  register(permission, callback, priority) {
+    hooks.addFilter(permission, callback, priority);
+    // Call all actions registered for permission.
+    // This will call an action for all AuthPoints with this permission
+    // and have them update state by running a new filter
+    // (which will include the newly added filter)
+    hooks.doAction(permission);
+  },
+
+  /**
+   * Unregisters package for updates on this AuthPoint permission
+   * @param  {String} permission listener is registered with
+   * @param  {Function} callback registered with listener
+   */
+  unregister(permission, callback) {
+    hooks.removeFilter(permission, callback);
+    // Have AuthPoints run filter again so removed filter will be excluded
+    hooks.doAction(permission);
+  },
+
+  // AuthPoint API
+  /**
+   * Register for changes in AuthService specific to the given permission.
+   * Whenever AuthService has a change it invoke callback
+   * so we can get the newly filtered authorized value.
+   * @param {String} permission to listen for events
+   * @param {Function} callback to call when event fires
+   * @param {Number} priority of callback [-Infinity; Infinity]
+   */
+  addListener(permission, callback, priority) {
+    hooks.addAction(permission, callback, priority);
+  },
+
+  /**
+   * Removes listener for changes in AuthService specific to the given
+   * permission.
+   * @param  {String} permission listener is registered with
+   * @param  {Function} callback registered with the permission
+   */
+  removeListener(permission, callback) {
+    hooks.removeAction(permission, callback);
+  },
+
+  /**
+   * Function to retrieve whether user is authorized for a given permission
+   * @param  {String} permission
+   * @param  {Boolean} defaultValue if no filter modifies value,
+   * this will be returned
+   * undefined or null
+   * @param  {Object} props to pass to filter function
+   * @return  {Boolean} wether the component is authorized or not
+   */
+  isAuthorized(permission, defaultValue, props) {
```

Comment:
Wonder why we need to pass all of these parameters, shouldn't it be enough to if a user has the required permissions?